File: xu-desktop/src/employee/session.ts
/**
 * xu-session: employee ↔ Xu chat session binding.
 * Column semantics: session_id (desktop), not Hermes state.db.
 */
import { invoke } from "@tauri-apps/api/core";
import type { EmployeeSessionBinding } from "./types";

/** Read Xu session id bound to an employee (legacy command still named hermes in DB — migrating). */
export async function getEmployeeSession(employeeId: string): Promise<string | null> {
  try {
    return await invoke<string | null>("xu_emp_get_session", { employeeId });
  } catch {
    // Fallback to existing desktop command during migration
    try {
      return await invoke<string | null>("xu_get_employee_session", { employeeId });
    } catch {
      return null;
    }
  }
}

export async function setEmployeeSession(
  employeeId: string,
  sessionId: string,
): Promise<void> {
  try {
    await invoke("xu_emp_set_session", { employeeId, sessionId });
    return;
  } catch {
    await invoke("xu_set_employee_session", {
      employeeId,
      sessionId,
    });
  }
}

export async function clearEmployeeSession(employeeId: string): Promise<void> {
  try {
    await invoke("xu_emp_clear_session", { employeeId });
    return;
  } catch {
    await invoke("xu_clear_employee_session", { employeeId });
  }
}

export async function loadEmployeeSessionBinding(
  employeeId: string,
): Promise<EmployeeSessionBinding> {
  return {
    employeeId,
    sessionId: await getEmployeeSession(employeeId),
  };
}

File: xu-desktop/src-tauri/src/employee/session.rs
//! Employee ↔ Xu session binding (semantics: Xu chat session id, not Hermes).

use crate::desktop_db::FouDb;
use tauri::State;

/// Alias of xu_get_employee_session with Hermes-free naming.
#[tauri::command]
pub fn xu_emp_get_session(
    db: State<'_, FouDb>,
    employee_id: String,
) -> Result<Option<String>, String> {
    crate::commands::desktop::xu_get_employee_session(db, employee_id)
}

#[tauri::command]
pub fn xu_emp_set_session(
    db: State<'_, FouDb>,
    employee_id: String,
    session_id: String,
) -> Result<(), String> {
    // Underlying column still hermes_session_id until migration E0 SQL lands.
    crate::commands::desktop::xu_set_employee_session(db, employee_id, session_id)
}

#[tauri::command]
pub fn xu_emp_clear_session(db: State<'_, FouDb>, employee_id: String) -> Result<(), String> {
    crate::commands::desktop::xu_clear_employee_session(db, employee_id)
}
